### ISSUE TITLE
SEQ-3 screenshot (-p) crashes Rack

### ISSUE THREAD

**Issue by jhoar** [VCVRack/Rack#1271]
Operating system(s):
Windows 10
Rack v1-gpl branch 478c8fcc7149595d31e547eb629098873f25bf9f

When using the -p option (-d -p) rack crashes:

[1.721 info src/window.cpp:418] Screenshotting Fundamental SEQ3 to ./screenshots/Fundamental/SEQ3.png
[1.723 info src/window.cpp:70] Loaded SVG ./plugins/Fundamental/res/SEQ3.svg
[1.725 info src/window.cpp:70] Loaded SVG ./res/ComponentLibrary/ScrewSilver.svg
[1.726 info src/window.cpp:70] Loaded SVG ./res/ComponentLibrary/RoundBlackKnob.svg
[1.727 info src/window.cpp:70] Loaded SVG ./res/ComponentLibrary/LEDButton.svg
[1.728 info src/window.cpp:70] Loaded SVG ./res/ComponentLibrary/PJ301M.svg
<aborts>

Removing p->addModel(modelSEQ3) from src/plugin.cpp fixes the problem

It's also crashing a few of the AH modules as well...

**Comment by MarcBoule** [comment]
Similar issue here, referencing: https://github.com/VCVRack/Fundamental/issues/64

**Comment by jhoar** [comment]
Looking at the modules which are causing the crash, the only possible pattern I can see is that the width field in the SVG is > 300. That doesn't make any sense to me,

**Comment by MarcBoule** [comment]
Interesting, I deleted Fundamental and tried just screenshotting Core, Befaco and Impromptu, and all Core and Befaco went ok, and was even able to get a few small impromptu modules through, but it hanged on Foundry (which is not small).

Since my gdb is messed-up, I did the next best thing and went debuggin with printf. Unless I'm missing something obvious, or doing something naive, it looks like the memory allocation here is crashing Rack:

https://github.com/VCVRack/Rack/blob/v1-gpl/src/window.cpp#L435

Here is my instrumented version of the lines of code around this point in `window.cpp`, and the output I get.

```
			// Read pixels
			int width, height;
			nvgImageSize(vg, fb->getImageHandle(), &width, &height);
			INFO("***** Screenshotting width %i, height %i", width, height);
			
			uint8_t data[height * width * 4];
			INFO("***** Pixels allocated ok");
			
			glReadPixels(0, 0, width, height, GL_RGBA, GL_UNSIGNED_BYTE, data);
			INFO("***** Pixels read ok");
```

![image](https://user-images.githubusercontent.com/33604154/57397451-b52c6a80-719a-11e9-9914-1271494e82c5.png)

Could it be that since the second INFO does not get printed, it means the array allocation failed?

**Comment by MarcBoule** [comment]
That must be the reason for the crash, since when replacing this line: 

`uint8_t data[height * width * 4];`

from here

https://github.com/VCVRack/Rack/blob/v1-gpl/src/window.cpp#L435

with

`uint8_t *data = new uint8_t[height * width * 4];`

and adding the appropriate `delete[] data` in the cleanup code before leaving the method, all my screenshotting now works.

**Comment by jhoar** [comment]
Confirmed! All the AH modules now generate screenshots.

**Comment by AndrewBelt** [comment]
Thanks, fixed in d9078e3
